test/lshr_shl_i1_regularize.ll

```diff
@@ -0,0 +1,40 @@
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv -s %t.bc -o %t.reg.bc
+; RUN: llvm-dis %t.reg.bc -o - | FileCheck --check-prefix=CHECK-LLVM %s
+
+target triple = "spir64-unknown-unknown"
+
+@G.0 = addrspace(1) global i1 false
+@G.1 = addrspace(1) global i1 true
+
+define spir_func void @test_lshr_i1(i1 %a, i1 %b) {
+entry:
+  %0 = lshr i1 %a, %b
+; CHECK-LLVM: [[AI32_0:%[0-9]+]] = select i1 %a, i32 1, i32 0
+; CHECK-LLVM: [[BI32_0:%[0-9]+]] = select i1 %b, i32 1, i32 0
+; CHECK-LLVM: [[LSHRI32:%[0-9]+]] = lshr i32 [[AI32_0]], [[BI32_0]]
+; CHECK-LLVM: [[TRUNC_0:%[0-9]+]] = icmp ne i32 [[LSHRI32]], 0
+  %1 = zext i1 %0 to i32
+  %2 = zext i1 %0 to i32
+; CHECK-LLVM-NOT zext
+; CHECK-LLVM-NOT select
+  store i1 %0, ptr addrspace(1) @G.0, align 1
+; CHECK-LLVM: store i1 [[TRUNC_0]], ptr addrspace(1) @G.0, align 1
+  ret void
+}
+
+define spir_func void @test_shl_i1(i1 %a, i1 %b) {
+entry:
+  %0 = shl i1 %a, %b
+; CHECK-LLVM: [[AI32_1:%[0-9]+]] = select i1 %a, i32 1, i32 0
+; CHECK-LLVM: [[BI32_1:%[0-9]+]] = select i1 %b, i32 1, i32 0
+; CHECK-LLVM: [[XORI32:%[0-9]+]] = xor i32 [[AI32_1]], [[BI32_1]]
+; CHECK-LLVM: [[TRUNC_1:%[0-9]+]] = icmp ne i32 [[XORI32]], 0
```

Comment:
Is shl i1 equivalent to xor?  If %a==0, shifting should produce a 0, but consider when %a==0 and %b==1:

%xor = xor i32 0, 1   ; produces 1
icmp ne i32 %xor, 0  ; produces 1

Thus the new code sequence produces a 1 when it should produce a 0.

Reply:
Left shift of bit:
```
0 << 0 = 0
1 << 0 = 1
0 << 1 = 1
1 << 1 = 0
```
which is the same for xor operation.

Logical right shift of bit:
```
0 >> 0 = 0
1 >> 0 = 1
0 >> 1 = 0
1 >> 1 = 0
```
I don't think that I made a mistake but maybe I'm missing something.

Reply:
> 0 << 1 = 1

I don't think this is correct.  Shifting 0 by anything produces a 0.

Reply:
The correct results are the same for both:

0 >> 0 = 0
1 >> 0 = 1
0 >> 1 = 0
1 >> 1 = 0


0 << 0 = 0
1 << 0 = 1
0 << 1 = 0
1 << 1 = 0

Reply:
Nice catch. Thank you very much.